File: client/src/routes/Awards/awards.js
import { Icon } from "semantic-ui-react";
import React from "react";
import ReactMarkdown from "react-markdown";

import SectionList from "../../components/SectionList";
import { awardsType } from "../../api/resume";

const Awards = ({ awards }) => {
  return (
    <div>
      {awards.map(award => (
        <SectionList
          key={award.title}
          title={
            <span className="title">
              <Icon name="star" />
              {award.title}
            </span>
          }
          subtitle={
            <span className="subtitle">
              <Icon name="thumbs up" />
              {award.awarder}
              <Icon className="ml-50" name="calendar" />
              {award.date}
            </span>
          }
          text={
            <span className="section-text">
              <ReactMarkdown source={award.summary} />
            </span>
          }
        />
      ))}
    </div>
  );
};

Awards.defaultProps = {
  awards: []
};

Awards.propTypes = awardsType;

export default Awards;

File: server/src/app/types/subtypes/awards.go
package subtypes
import (
	"github.com/graphql-go/graphql"
)

var Awards = graphql.NewObject(graphql.ObjectConfig {
	Name: "Awards",
	Fields: graphql.Fields{
		"title": &graphql.Field{
			Type: graphql.String,
		},
		"date": &graphql.Field{
			Type: graphql.String,
		},
		"awarder": &graphql.Field{
			Type: graphql.String,
		},
		"summary": &graphql.Field{
			Type: graphql.String,
		},
	},
})